### PR TITLE
Fix missing notifications

### PR DESCRIPTION
**Pull request by benoitvidis**
# Description

Fixes some cases where Kuzzle DSL would send notifications to a subset of matching subscription rooms only.
In depth, the internal test table used by the DSL could assign the same index to different filters, making the notifier trigger only one notification for these.

The fix simplifies the way the test tables indices are computed upon a reinxing (following an unregistration).

# Related issue

* #822

### CODE REVIEW

File: lib/api/dsl/storage/index.js [hunk 1]
```diff
@@ -266,6 +266,124 @@ class Storage {
...
+    if (tt.removedConditions.array.length < tt.clength / 10) {
+      return;
+    }
+
+    // rebuild conditions index
+    const
+      conditionKeys = Object.keys(this.subfilters[index][collection]),
+      conditions = new Uint8Array(tt.conditions.length);
+
+    for (let i = 0; i < conditionKeys.length; i++) {
```
Comment (scottinet) [inline lib/api/dsl/storage/index.js]: `i` should be declared outside the for loop for better performances

File: lib/api/dsl/storage/index.js [hunk 2]
```diff
@@ -266,6 +266,124 @@ class Storage {
...
+
+      conditions[i] = sf.conditions.length;
+      sf.cidx = i;
+    }
+    tt.conditions = conditions;
+    tt.clength = conditionKeys.length;
+
+    // rebuild filter index pointers
+    if (tt.removedFiltersCount) {
+      for (let i = 0; i < Object.keys(this.filtersIndex[index][collection]).length; i++) {
```
Comment (scottinet) [inline lib/api/dsl/storage/index.js]: Same here

Comment (benoitvidis) [inline lib/api/dsl/storage/index.js]: @scottinet do we really care during unregistration?

Comment (scottinet) [inline lib/api/dsl/storage/index.js]: Unfortunately yes: unregistration and reindexation are both very time-consuming tasks, and these therefore should be considered critical code.

Comment (benoitvidis) [inline lib/api/dsl/storage/index.js]: ok. fixed.

**Pull request merged**